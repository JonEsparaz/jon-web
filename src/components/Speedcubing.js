import React from 'react';
import './Speedcubing.css';

function Speedcubing() {
  return (
    <div className="CubingContainer">
      <p></p>
      <a href="https://www.worldcubeassociation.org/persons/2013ESPA01">WCA records</a>
    </div>
  );
}

export default Speedcubing;
