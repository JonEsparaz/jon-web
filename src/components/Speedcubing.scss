@media (min-width: 768px) {
    .CubingContainer {
        width: 80vw;
        margin-left: 10vw;
        text-align: left;
        position: relative;
    }
    .YTimage {
        width: 26vw;
        cursor: pointer;
    }

    .YTtitle {
        font-weight: bold;
        width: 24vw;
    }

    .YTimage:hover {
        opacity: 0.5;
    }

    .YTgrid {
        display: grid;
        grid-template-columns: repeat(3,26vw);
        grid-template-rows: auto;
        column-gap: 1vw
    }

    .custom-modal {
        display: flex;
        text-align: center;
        justify-content: center;
    }

    .videoPlayer {
        width: 50vw;
        height: calc(50vw * 9/16);
    }

}

@media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape) {
    .CubingContainer {
        width: 80vw;
        margin-left: 10vw;
        text-align: left;
        position: relative;
    }
    .YTimage {
        width: 26vw;
        cursor: pointer;
    }

    .YTtitle {
        font-weight: bold;
        width: 24vw;
    }

    .YTimage:hover {
        opacity: 0.5;
    }

    .YTgrid {
        display: grid;
        grid-template-columns: repeat(3,26vw);
        grid-template-rows: auto;
        column-gap: 1vw
    }

    .custom-modal {
        display: flex;
        text-align: center;
        justify-content: center;
    }

    .videoPlayer {
        width: 50vw;
        height: calc(50vw * 9/16);
    }
}

@media (max-width: 767px) {
    .CubingContainer {
        width: 90vw;
        margin-left: 5vw;
        text-align: left;
        position: relative;
    }

    .YTimage {
        width: 90vw;
        cursor: pointer;
    }

    .YTtitle {
        font-weight: bold;
        width: 85vw;
    }

    .YTimage:hover {
        opacity: 0.5;
    }

    .YTgrid {
        display: grid;
        grid-template-columns: repeat(1,90vw);
        grid-template-rows: auto;
        row-gap: 5vw;
    }
    
    .custom-modal {
        display: flex;
        text-align: center;
        justify-content: center;
    }

    .videoPlayer {
        width: 50vw;
        height: calc(50vw * 9/16);
    }
}