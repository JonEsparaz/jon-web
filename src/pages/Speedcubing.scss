@import '../theme.scss';

.CubingContainer {
  font-family: $font;
  text-align: center;
}

.SpeedcubingText {
  text-align: left;
}

@media (min-width: 768px) {
  .CubingContainer {
    width: 80vw;
    margin-left: 10vw;
    min-height: calc(100vh - 96px - 159px - 10vw);
    margin-bottom: 5vw;
  }

  .SpeedcubingText {
    margin-bottom: 4vw;
  }

  .YTimage {
    width: 24vw;
    height: (9/16) * 24vw;
    cursor: pointer;
    opacity: 1;
    transition: opacity 0.3s ease-in-out;
  }

  .YTtitle {
    font-family: $font;
    font-size: 1.2vw;
    color: black;
    text-align: left;
  }

  .imageContainer {
    position: relative;
    text-align: center;
    display: block;
    border: none;
    outline: none !important;
    background: transparent;
    padding: 0;
  }

  .imageContainer:hover,
  .imageContainer:focus {
    cursor: pointer;
    .YTimage {
      opacity: 0.8;
    }
    .playArrow {
      opacity: 0.9;
    }
  }

  .YTgrid {
    display: grid;
    margin-bottom: 4vw;
    grid-template-columns: repeat(3, 26vw);
    gap: 1vw;
  }

  .playArrow {
    position: absolute;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    transition: opacity 0.3s ease-in-out;
  }

  .custom-modal {
    display: flex;
    text-align: center;
    justify-content: center;
    top: 10vw !important;
  }

  .videoPlayer {
    width: 50vw;
    height: calc(50vw * 9 / 16);
  }

  .SpeedcubingItem {
    padding: 1vw;
    box-shadow: $shadow;
  }
}

@media (max-width: 767px) {
  .CubingContainer {
    margin-left: 5vw;
    margin-right: 5vw;
    margin-bottom: 10vw;
  }

  .YTimage {
    width: 80vw;
    height: (9/16) * 80vw;
    cursor: pointer;
    opacity: 1;
    transition: opacity 0.3s ease-in-out;
  }

  .YTtitle {
    font-family: $font;
    font-size: 4vw;
    color: black;
    text-align: left;
  }

  .imageContainer {
    position: relative;
    text-align: center;
    display: block;
    border: none;
    outline: none !important;
    background: transparent;
    padding: 0;
  }

  .imageContainer:hover:focus {
    cursor: pointer;
    .YTimage {
      opacity: 0.8;
    }
    .playArrow {
      opacity: 0.9;
    }
  }

  .YTgrid {
    display: grid;
    grid-template-columns: repeat(1, 90vw);
    gap: 5vw;
  }

  .playArrow {
    position: absolute;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    transition: opacity 0.3s ease-in-out;
  }

  .custom-modal {
    display: flex;
    text-align: center;
    justify-content: center;
    top: 10vw !important;
  }

  .videoPlayer {
    width: 50vw;
    height: calc(50vw * 9 / 16);
  }

  .SpeedcubingItem {
    padding: 5vw;
    box-shadow: $shadow;
  }
}
