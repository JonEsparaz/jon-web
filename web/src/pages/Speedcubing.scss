.modal-body-custom {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.modal-custom {
  max-width: 70vw !important;
}

.video-player {
  width: 65vw;
  height: 9/16 * 65vw;
}

.image-container {
  position: relative;
  text-align: center;
  display: block;
  border: none;
  outline: none !important;
  background: transparent;
  padding: 0;

  &:focus,
  &:hover {
    cursor: pointer;
    .yt-image {
      opacity: 0.8;
    }
    .play-arrow {
      opacity: 0.9;
    }
  }
}

.yt-image {
  transition: opacity 0.3s ease-in-out;
}

.play-arrow {
  position: absolute;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
}

.modal-content {
  border-radius: 0;
}

@media (max-width: 767px) {
  .modal-custom {
    display: none !important;
  }
}
