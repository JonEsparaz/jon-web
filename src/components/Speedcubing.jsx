import React from 'react';
import { Modal } from '@material-ui/core';
import { isMobile } from 'react-device-detect';
import './Speedcubing.scss';

class Speedcubing extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      modal: false,
      video: null
    }
    this.videoData = [
      {id:"U5x6YN1k6T8", text:"First Place @ Newmarket Open 2019: 7.39 3x3 Rubik's Cube Average"}, 
      {id:"E5h2XIfgcow", text:"North American Record: 7.57 3x3 One-Handed Single"},
      {id:"aJKcaV2tP_o", text:"Second Place @ Canadian Championship 2019: 12.94 3x3 One-Handed Average"}
    ]
  }

  handleOpen(data) {
    if (isMobile) {
      window.location.href = `https://www.youtube.com/watch?v=uAAuNjNXaIM${data}`;
    } else {
      this.setState({
        modal: true,
        video: data
      })
    }
  };

  handleClose = () => {
    this.setState({
      modal: false
    })
  };

  videoModal() {
    return (
      <Modal
        open={this.state.modal}
        onClose={this.handleClose}
        className={"custom-modal"}
      >
        <iframe 
            className="videoPlayer"
            title="videoPlayer"
            src={`https://www.youtube.com/embed/${this.state.video}?autoplay=1`}
            frameBorder="0"
          />
      </Modal>
    )
  }

  render() {
    return (
      <div className="CubingContainer">
        {this.videoModal()}
        <h1 className="SpeedcubingH1">Speedcubing</h1>
        <div className="SpeedcubingText">Over the past {new Date().getFullYear()-2013} years, I have competed over 50 
        World Cube Association competitions. In 2018, I broke the North American Record for solving the Rubik's Cube one-handed with a time of 7.57 seconds. 
        Today, I am a Junior Delegate for the World Cube Association and help to organize and officiate competitions. 
        Check out my WCA profile <a href="https://www.worldcubeassociation.org/persons/2013ESPA01"> here</a>.</div>
        
        <div className="SpeedcubingH2">Check out some of my top solves</div>
        <div className="YTgrid">
          {this.videoData.map(item => {
              return (
                <div classname="YTitem">
                  <div className="imageContainer" onClick={()=>{this.handleOpen(item.id)}}>
                  <img className="YTimage" src={`https://img.youtube.com/vi/${item.id}/maxresdefault.jpg`} alt="youtube thumbnail" ></img>
                  <img className="playArrow" src='/svg/play_arrow-white.svg' alt=""></img>
                  </div>
                  <div className="YTtitle">{item.text}</div>
                </div>
              )
            }
          )}
        </div>
      </div>
    )
  }

}

export default Speedcubing;
