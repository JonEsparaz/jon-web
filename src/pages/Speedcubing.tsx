import React, { useState } from 'react';
import { Modal } from '@material-ui/core';
import { isMobile } from 'react-device-detect';
import Menu from '../components/Menu';
import Footer from '../components/Footer';
import './Speedcubing.scss';

export default function Speedcubing(): JSX.Element {
  const [video, setVideo] = useState('');
  const videoData = [
    {
      id: 'E5h2XIfgcow',
      text: 'North American Record: 7.57 3x3 One-Handed Solve',
    },
    {
      id: 'U5x6YN1k6T8',
      text: "1st Place @ Newmarket Open 2019: 7.39 3x3 Rubik's Cube Average",
    },
    {
      id: 'aJKcaV2tP_o',
      text:
        '2nd Place @ Canadian Championship 2019: 12.94 3x3 One-Handed Average',
    },
  ];

  const handleOpen = (data: string) => {
    if (isMobile) {
      window.location.href = `https://www.youtube.com/watch?v=${data}`;
    } else {
      setVideo(data);
    }
  };

  const videoModal = () => {
    return (
      <Modal
        open={Boolean(video)}
        onClose={() => setVideo('')}
        className="custom-modal"
      >
        <iframe
          className="videoPlayer"
          title="Video Player"
          src={`https://www.youtube.com/embed/${video}?autoplay=1`}
          frameBorder="0"
        />
      </Modal>
    );
  };

  return (
    <div className="page-wrapper">
      <Menu mode="dark" />
      <div className="center-wrapper page-body">
        <div className="CubingContainer">
          {videoModal()}
          <h2 className="Header2 Big">Speedcubing</h2>
          <div className="Text1 SpeedcubingText">
            For the past {new Date().getFullYear() - 2013} years, I have
            competed in World Cube Association competitions. In 2018, I broke
            the North American Record for solving the Rubik&apos;s Cube
            one-handed with a time of 7.57 seconds. At the time, this ranked me
            second in the world. Today, I organize and officiate competitions as
            a Junior Delegate for the World Cube Association.
          </div>
          <h2 className="Header2">Highlight Reel</h2>
          <div className="YTgrid">
            {videoData.map((item) => {
              return (
                <div className="SpeedcubingItem" key={item.id}>
                  <button
                    type="button"
                    aria-label={`Open video: ${item.text}`}
                    className="imageContainer"
                    onClick={() => {
                      handleOpen(item.id);
                    }}
                  >
                    <img
                      className="YTimage"
                      alt="YouTube thumbnail"
                      src={`https://img.youtube.com/vi/${item.id}/maxresdefault.jpg`}
                    />
                    <img
                      className="playArrow"
                      src="/svg/play_arrow-white.svg"
                      alt=""
                    />
                  </button>
                  <div className="YTtitle">{item.text}</div>
                </div>
              );
            })}
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
}
