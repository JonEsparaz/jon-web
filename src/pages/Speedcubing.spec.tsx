import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import * as rdd from 'react-device-detect';
import Speedcubing from './Speedcubing';

const videoData = [
  {
    id: 'E5h2XIfgcow',
    text: 'North American Record: 7.57 3x3 One-Handed Solve',
  },
  {
    id: 'U5x6YN1k6T8',
    text: "1st Place @ Newmarket Open 2019: 7.39 3x3 Rubik's Cube Average",
  },
  {
    id: 'aJKcaV2tP_o',
    text:
      '2nd Place @ Canadian Championship 2019: 12.94 3x3 One-Handed Average',
  },
];

describe('speedcubing page', () => {
  test('each video renders', () => {
    const { container, getByText } = render(
      <MemoryRouter>
        <Speedcubing />
      </MemoryRouter>,
    );

    videoData.forEach((video) => {
      expect(getByText(video.text)).toBeTruthy();
    });

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('open video, desktop', () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    rdd.isMobile = false;
    const { getByLabelText, queryByRole } = render(
      <MemoryRouter>
        <Speedcubing />
      </MemoryRouter>,
    );
    expect(queryByRole('presentation')).toBeFalsy();

    const button = getByLabelText(`Open video: ${videoData[0].text}`);

    fireEvent.click(button);
    expect(queryByRole('presentation')).toBeTruthy();
    jest.clearAllMocks();
  });

  test('open video, mobile', () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    rdd.isMobile = true;
    const { queryByRole, getByLabelText } = render(
      <MemoryRouter>
        <Speedcubing />
      </MemoryRouter>,
    );

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    delete window.location;

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    window.location = { assign: jest.fn() };

    expect(queryByRole('presentation')).toBeFalsy();

    const button = getByLabelText(`Open video: ${videoData[0].text}`);
    fireEvent.click(button);

    expect(queryByRole('presentation')).toBeFalsy();
    expect(window.location.assign).toHaveBeenCalledWith(
      `https://www.youtube.com/watch?v=${videoData[0].id}`,
    );
  });
});
