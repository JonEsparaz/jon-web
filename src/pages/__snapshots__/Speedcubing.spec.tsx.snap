// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`speedcubing page each video renders 1`] = `"<div class=\\"page-wrapper\\"><div class=\\"MenuContainer\\" style=\\"position: relative; z-index: 9999;\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo.png\\" style=\\"height: 80px; margin-right: 2vw;\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider  bl-menu\\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><a class=\\"navlink-custom\\" href=\\"/speedcubing\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z\\"></path></svg><span class=\\"Underline\\">Speedcubing</span></a><a class=\\"navlink-custom\\" href=\\"/contact\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z\\"></path></svg><span class=\\"Underline\\">Contact</span></a></ul></div></nav></div><div class=\\"center-wrapper page-body\\"><div class=\\"CubingContainer\\"><h2 class=\\"Header2 Big\\">Speedcubing</h2><div class=\\"Text1 SpeedcubingText\\">For the past 8 years, I have competed in World Cube Association competitions. In 2018, I broke the North American Record for solving the Rubik's Cube one-handed with a time of 7.57 seconds. At the time, this ranked me second in the world. Today, I organize and officiate competitions as a Junior Delegate for the World Cube Association.</div><h2 class=\\"Header2\\">Highlight Reel</h2><div class=\\"YTgrid\\"><div class=\\"SpeedcubingItem\\"><button type=\\"button\\" aria-label=\\"Open video: North American Record: 7.57 3x3 One-Handed Solve\\" class=\\"imageContainer\\"><img class=\\"YTimage\\" alt=\\"YouTube thumbnail\\" src=\\"https://img.youtube.com/vi/E5h2XIfgcow/maxresdefault.jpg\\"><img class=\\"playArrow\\" src=\\"/svg/play_arrow-white.svg\\" alt=\\"\\"></button><div class=\\"YTtitle\\">North American Record: 7.57 3x3 One-Handed Solve</div></div><div class=\\"SpeedcubingItem\\"><button type=\\"button\\" aria-label=\\"Open video: 1st Place @ Newmarket Open 2019: 7.39 3x3 Rubik's Cube Average\\" class=\\"imageContainer\\"><img class=\\"YTimage\\" alt=\\"YouTube thumbnail\\" src=\\"https://img.youtube.com/vi/U5x6YN1k6T8/maxresdefault.jpg\\"><img class=\\"playArrow\\" src=\\"/svg/play_arrow-white.svg\\" alt=\\"\\"></button><div class=\\"YTtitle\\">1st Place @ Newmarket Open 2019: 7.39 3x3 Rubik's Cube Average</div></div><div class=\\"SpeedcubingItem\\"><button type=\\"button\\" aria-label=\\"Open video: 2nd Place @ Canadian Championship 2019: 12.94 3x3 One-Handed Average\\" class=\\"imageContainer\\"><img class=\\"YTimage\\" alt=\\"YouTube thumbnail\\" src=\\"https://img.youtube.com/vi/aJKcaV2tP_o/maxresdefault.jpg\\"><img class=\\"playArrow\\" src=\\"/svg/play_arrow-white.svg\\" alt=\\"\\"></button><div class=\\"YTtitle\\">2nd Place @ Canadian Championship 2019: 12.94 3x3 One-Handed Average</div></div></div></div></div><div class=\\"FooterContainer\\"><a aria-label=\\"Twitter\\" href=\\"https://twitter.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeLarge\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" style=\\"color: black;\\"><path d=\\"M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z\\"></path></svg></a><a aria-label=\\"GitHub\\" href=\\"https://github.com/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeLarge\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" style=\\"color: black;\\"><path d=\\"M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3\\"></path></svg></a><a aria-label=\\"YouTube\\" href=\\"https://www.youtube.com/c/JonEsparaz\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeLarge\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" style=\\"color: black;\\"><path d=\\"M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z\\"></path></svg></a><a aria-label=\\"LinkedIn\\" href=\\"https://www.linkedin.com/in/jonathanesparaz/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"footer-link\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeLarge\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" style=\\"color: black;\\"><path d=\\"M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z\\"></path></svg></a><div style=\\"margin-top: 20px; color: black;\\"> Â© 2021 Jonathan Esparaz</div></div></div>"`;
